### PR TITLE
catch exceptions by reference and print relevent information

### PR DESCRIPTION
**Pull request by SteveMacenski**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | https://github.com/ros-planning/navigation2/issues/659 |
| Primary OS tested on | (N/A |
| Robotic platform tested on | N/A |

---

## Description of contribution in a few bullet points

* Catch exception by references across stack
* print exceptions as needed

---

## Future work that may be required in bullet points

---

I'd like to request maintainer: <blank> to review this PR.

### CODE REVIEW

File: nav2_map_server/src/map_server.cpp [hunk 1]
```diff
@@ -76,15 +76,16 @@ void MapServer::getParameters()
       map_filename_ = std::string(dirname(fname_copy)) + '/' + map_filename_;
       free(fname_copy);
     }
-  } catch (YAML::Exception) {
-    std::string msg = "'" + yaml_filename_ + "' does not contain an image tag or it is invalid";
+  } catch (YAML::Exception & e) {
+    std::string msg = "'" + yaml_filename_ + \
```
Comment (crdelsey) [inline nav2_map_server/src/map_server.cpp]: The backslash at the end of the line shouldn't be necessary.

Comment (SteveMacenski) [reply nav2_map_server/src/map_server.cpp]: Shouldnt be necessary, but a typical IDE when that is present will comment out all lines that are continued with the `\` so it helps in debugging.

Comment (SteveMacenski) [reply nav2_map_server/src/map_server.cpp]: I can remove if you like.

Comment (crdelsey) [reply nav2_map_server/src/map_server.cpp]: I dislike line continuation characters but if it plays nicer with the tools, that's a good reason to keep them. 

However, the logical conclusion would be every long line that is split should have a line continuation character at the end. We have a lot of them. Is that really where we want to go with the code?

Comment (SteveMacenski) [reply nav2_map_server/src/map_server.cpp]: Generally I do put them everywhere, but I did this one pretty fast, I'll just remove it to keep it all consistent

**Comment by SteveMacenski** [comment]
Not sure why the debug build failed, looks like some caching issue in Circle. I'm going to add an extraneous line to trigger a total new build. Rerunning the workflow only did the failed bit which I think had the caching problem.

**Comment by crdelsey** [comment]
@SteveMacenski  FYI - Since you have full write privileges to this repo, you can just rerun the build. Click on the build log, log in to travis or circleCI with your github account, and then you should have a restart build option

**Comment by SteveMacenski** [comment]
Hi,

I saw a button for rerun workflow, but CircleCI automatically cached the successfully built parts and then when it came to the last section, it was looking for a file that didnt exist. I couldnt find a way to rerun the entire job, Travis I know how to rerun but I guess I need to spend some more time with Circle to find those options.

**Comment by SteveMacenski** [comment]
@mhpanah @mjeronimo can I get a second review? This is a small change

**Comment by SteveMacenski** [comment]
Checking back on this -- very small change that would help build on 18.10 -- can I get a second OK to merge? @mhpanah @mjeronimo

**Comment by crdelsey** [comment]
Everyone is focussed on getting the lifecycle branch integrated with master right now. There's a reluctance to add anything else to master to avoid additional rebase conflicts, unless it is necessary to unbreak the build.

This one is in a grey area with respect to unbreaking the build. It unbreaks the build for 18.10 but isn't needed for the official build.

Because this touches amcl, which is probably the worst rebase conflict on the lifecycle branch, and because master is so broken all the time for everybody due to the constant API changes, I'm inclined to want to hold onto this until lifecycle is merged to master. Are you OK with that?

**Comment by SteveMacenski** [comment]
@crdelsey that works, I'm just starting to get a few PRs backed up and I like to clean house before moving forward. What's the timeline on the lifecycle diff landing?

**Comment by crdelsey** [comment]
@SteveMacenski I think lifecycle will be merged towards the end of this week.

**Comment by SteveMacenski** [comment]
Sounds good

**Comment by crdelsey** [comment]
@SteveMacenski The lifecycle branch has been merged to master now.

**Comment by crdelsey** [comment]
@SteveMacenski Are you available to rebase this? If not, I can tackle that.

**Comment by crdelsey** [comment]
There's overlap with #871. It catches exceptions by reference, but doesn't add the extra error output.

**Pull request merged**